src/utils/password.ts

```diff
@@ -1,11 +1,11 @@
-import { sha256 } from 'hash.js';
+import argon2 from 'argon2';
 
-const hashFunction = (secret: string) => {
-  return sha256().update(secret).digest('hex');
+const hashFunction = async (secret: string) => {
+  return argon2.hash(secret, { type: argon2.argon2id });
```

Comment:
I think this also need error handler since it's a promise (the same with `checkHash()`)